test/view_component/integration_test.rb

```diff
@@ -475,4 +475,16 @@ class IntegrationTest < ActionDispatch::IntegrationTest
       assert_select "input[value=?]", "Send this form!"
     end
   end
+
+  test "Rails fragment cache gets flushed correctly when the component template changes" do
+    get "/fragment_cache"
+    assert_response :success
+    assert_select("div", "Foo\n  bar")
+    assert_select("div.changed", false)
+
+    modify_file "app/components/erb_component.html.erb", "<div class='changed'>Hello World</div>" do
```

Comment:
I don't think this is correct in the context of fragment caching. 

The template is cached using a key, when the key changes the cache block is executed again and cached under the new key. The old key will be flushed, in time, depending on the eviction algorithm using the cache itself.

This example is rather showing the scenario where a developer changes the production code. Which is a different problem, one should decide to either update the cache key construction before releasing the changes or maybe using an Active Record object that responds to cache_key_with_version.

Moreover I think ViewComponent already play nice with Rails' fragment caching, see this example https://github.com/github/view_component/blob/master/test/view_component/integration_test.rb#L159 using Rails' `cache_if`.

Reply:
Well nevermind, I just realized this is a valid case when the cache entry can't be manually invalidated or changed. 

I'll try to lend a hand into investigating how to solve the issue.